### PR TITLE
Adds support for integrity checking of LCOW layers

### PR DESCRIPTION
**Pull request by anmaxvl**
A new feature has been added to hcsshim where read-only LCOW layers
can be mounted with integrity protection. This is done by reading
the hash device information from layer VHD and passing it to the
LCOW UVM, where device mapper checks the integrity via dm-verity
Linux kernel feature. Layout on disk is expected to be:

```
|Byte 0   |Byte size(ext4)      |Byte size(ext4)+size(hash device)|
|ext4 data|dm-verity hash device|VHD footer                       |
```

This PR adds ability to append hash device data during LCOW image
extraction. The additional metadata to tar2ext4 converter is passed
via diff.ApplyConfig.ProcessorPayloads.

Update hcsshim vendor to v0.9.0 

Signed-off-by: Maksim An <maksiman@microsoft.com>

### CODE REVIEW

**Review by dcantah — APPROVED** [review]
lgtm

**Comment by anmaxvl** [comment]
converting to draft, since the approach is being re-evaluated.

**Comment by anmaxvl** [comment]
@dcantah can you revisit this? having the client option in our fork makes more sense vs having it upstreamed like I was going to do in this PR https://github.com/containerd/containerd/pull/6000

**Comment by anmaxvl** [comment]
@dcantah @ambarve @katiewasnothere @kevpar @helsaawy @msscotb PTAL

**Review by msscotb — APPROVED** [review]
lgtm

**Review by msscotb — APPROVED** [review]

File: vendor.conf [hunk 1]
```diff
@@ -28,7 +28,7 @@ github.com/imdario/mergo                            v0.3.7
 github.com/konsorten/go-windows-terminal-sequences  v1.0.3
 github.com/matttproud/golang_protobuf_extensions    v1.0.1
 github.com/Microsoft/go-winio                       v0.4.14
-github.com/Microsoft/hcsshim                        v0.8.10
+github.com/Microsoft/hcsshim                        77c0270424049ab4c850c076601add11882e7c1e
```
Comment (dcantah) [inline vendor.conf:31]: Do we not have a tag that contains the work?

Comment (anmaxvl) [reply vendor.conf:31]: 0.9.x should have the appending of layer hashes, however, that implementation reads entire layer in memory and doesn't support passing an io.Reader. If we cut a 0.10 release, I'd be happy to use that, but I don't think it's happening yet?

Comment (dcantah) [reply vendor.conf:31]: Ahhh, gotcha. Going by commit is likely fine for now then. Whenever we cut a tag we can update

**Review by dcantah — APPROVED** [review]

**Pull request merged**